premake5.lua

```diff
@@ -87,7 +87,7 @@ filter("configurations:Release")
     "NDEBUG",
     "_NO_DEBUG_HEAP=1",
   })
-  optimize("Speed")
+  optimize("Full")
```

Comment:
What's the difference between Speed and Full? Premake documentation doesn't say much https://premake.github.io/docs/optimize

Reply:
This change should be reverted.

Reply:
I changed the title to "[MISC] Preparation for C++20", is this okay?

Reply:
Sorry, but can you change the commit name to something like: ``[Base] Fixed compilation error in math.h``

I tried porting Xenia once to C++20 for some nice features, but it is painful due to utf and more